man/fi_peer.3.md

```diff
@@ -446,15 +451,22 @@ call to the owner's queue_msg/tag. The get and queue messages should be serializ
 When the owner gets a matching receive for the queued unexpected message, it will
 call the peer's start function to notify the peer of the updated rx_entry (or the
 peer's discard function if the message is to be discarded)
-(TBD: The peer may need to update the src addr if the remote endpoint is
-inserted into the AV after the message has been received.)
 
 # fi_ops_srx_owner::queue_msg() / queue_tag()
 
 Called by the peer to queue an incoming unexpected message to the srx. Once it
 gets queued by the peer, the owner is responsible for starting it once it gets
 matched to a receive buffer, or discard it if needed.
 
+## fi_ops_srx_owner::queue_msg_progress() / queue_tag_progress();
+
+Queues an unexpected message with a progress function call. This progress
+function gets called by the owner on every post receive call to allow the peer
+to update any fields, specifically the address field. This allows the peer to
+queue a message from an unknown peer and update it later once the fi_addr is
+known. Note that messages in the progress queue can still be matched to receives
+posted with FI_ADDR_UNSPEC.
```

Comment:
I need more time to think about this.  But calling the progress function every progress call is inefficient.  Unless an unmatched directed receive has been posted, there's nothing to do.  And even in that case, unless some address has been inserted into the AV, no progress would be made.